scheduler/cloudformation/template.go

```diff
@@ -187,19 +189,47 @@ func (t *EmpireTemplate) Build(app *scheduler.App) (*troposphere.Template, error
 }
 
 func (t *EmpireTemplate) addScheduledTask(tmpl *troposphere.Template, app *scheduler.App, p *scheduler.Process) (taskDefinition string) {
+	taskDefinitionType := taskDefinitionType(app)
+
 	key := processResourceName(p.Type)
 
+	cd := t.ContainerDefinition(app, p)
 	// The task definition that will be used to run the ECS task.
-	containerDefinition := containerDefinition(t.ContainerDefinition(app, p))
 	taskDefinition = fmt.Sprintf("%sTaskDefinition", key)
-	tmpl.Resources[taskDefinition] = troposphere.Resource{
-		Type: "AWS::ECS::TaskDefinition",
-		Properties: map[string]interface{}{
-			"ContainerDefinitions": []interface{}{
-				containerDefinition,
+	containerDefinition := containerDefinition(cd)
+
+	taskDefinitionProperties := map[string]interface{}{
+		"Volumes": []interface{}{},
+	}
+	if taskDefinitionType == "Custom::ECSTaskDefinition" {
```

Comment:
it looks like this block is the same as: https://github.com/remind101/empire/pull/935/files#diff-ae0d1a0ca421a9a4d5510666ef5b1c5cR411, is it worth abstracting into a function?